src/library/scala/collection/mutable/PriorityQueue.scala

```diff
@@ -254,12 +320,24 @@ sealed class PriorityQueue[A](implicit val ord: Ordering[A])
    *
    *  @return  a priority queue with the same elements.
    */
-  override def clone(): PriorityQueue[A] = new PriorityQueue[A] ++= this.iterator
+  override def clone(): PriorityQueue[A] = {
+    val pq = new PriorityQueue[A]
+    for (i <- 1 to this.length) pq.unsafeAdd(resarr(i))
+    pq
+  }
 }
 
 
 object PriorityQueue extends OrderedTraversableFactory[PriorityQueue] {
-  def newBuilder[A](implicit ord: Ordering[A]) = new PriorityQueue[A]
+  def newBuilder[A](implicit ord: Ordering[A]): Builder[A, PriorityQueue[A]] = {
+    new Builder[A, PriorityQueue[A]] {
+      val pq = new PriorityQueue[A]
+      def +=(elem: A): this.type = { pq.unsafeAdd(elem); this }
+      def result(): PriorityQueue[A] = { pq.heapify(1); pq }
```

Comment:
Can you decide whether this builder should be reusable or not (see the new ReusableBuilder trait)?  As written, the builder can be used once efficiently, but can still be added to and regenerated a little less efficiently (because heapify is called from 1, not only on the new elements).  Since most builder-use is single shot, this is probably the better way to go (as opposed to maintaining a list of number of elements stored since the last call to result).  In any case, this seems reusable, so maybe you can tag it as such.